recipe/build.sh

```diff
@@ -5,3 +5,18 @@ $PYTHON -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
 rm -rf $PREFIX/man
 rm -f $PREFIX/bin/spyder_win_post_install.py
 rm -rf $SP_DIR/Sphinx-*
+
+# Create the Menu directory
+mkdir -p "${PREFIX}/Menu"
+
+# Copy menu.json template
+cp "${RECIPE_DIR}/spyder-menu.json" "${PREFIX}/Menu/spyder-menu.json"
+
+# Copy application icons
+if [[ $OSTYPE == "darwin"* ]]; then
+    cp "${RECIPE_DIR}/spyder.icns" "${PREFIX}/Menu/spyder.icns"
+    # Update version in menu.json
+    sed -i "" -e "s/__PKG_VERSION__/${PKG_VERSION}/g" "${PREFIX}/Menu/spyder-menu.json"
```

Comment:
It seems `sed` is not available in the Mac arm64 slots and that's why the build fails at this point.

As a solution, we should add [`sed`](https://github.com/conda-forge/sed-feedstock) as a dependency during the build phase.

Reply:
That's interesting, but I'll add it.

Reply:
I've discovered the root issue. `sed` is indeed available in the macOS arm64 slot, but it is the GNU version rather than the BSD version. This appears to be unique to CI since this is not the case on my M1 iMac. Checking on the pipeline build steps:
<details><summary>osx_arm64 build</summary>

```
$ what /usr/bin/sed
/usr/bin/sed
	PROGRAM:sed  PROJECT:text_cmds-106
	PROGRAM:sed  PROJECT:text_cmds-106
$ [[ $(sed --version 2> /dev/null) ]] && sed --version || echo "sed: bsd"
sed (GNU sed) 4.7
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Jay Fenlason, Tom Lord, Ken Pizzini,
Paolo Bonzini, Jim Meyering, and Assaf Gordon.
GNU sed home page: <https://www.gnu.org/software/sed/>
General help using GNU software: <https://www.gnu.org/gethelp/>.
E-mail bug reports to: <bug-sed@gnu.org>.
```
</details>
<details><summary>osx_64 build</summary>

```
$ what /usr/bin/sed
/usr/bin/sed
	PROGRAM:sed  PROJECT:text_cmds-106
	PROGRAM:sed  PROJECT:text_cmds-106
$ [[ $(sed --version 2> /dev/null) ]] && sed --version || echo "sed: bsd"
sed: bsd
```
</details>

See [spyder-feedstock 20231226.5](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=849732&view=results), and the logs for the osx_64 vs osx_arm64. The relevant distinction is that there are differences in the implementation of the in-place flag (`-i`). However, we don't need edit in place in this circumstance, so I bypassed it altogether.

What is interesting is that both osx_64 and osx_arm64 use the same platform image 🤷🏼. 
<details><summary>osx_64</summary>

```
Starting: Initialize job
Agent name: 'Azure Pipelines 19'
Agent machine name: 'Mac-1703606051358'
Current agent version: '3.232.0'
Operating System
macOS
11.7.10
20G1427
Runner Image
Image: macos-11
Version: 20231216.1
Included Software: https://github.com/actions/runner-images/blob/macOS-11/20231216.1/images/macos/macos-11-Readme.md
Image Release: https://github.com/actions/runner-images/releases/tag/macOS-11%2F20231216.1
Runner Image Provisioner
2.0.321.1
Current image version: '20231216.1'
```
</details>
<details><summary>osx_arm64</summary>

```
Starting: Initialize job
Agent name: 'Azure Pipelines 38'
Agent machine name: 'Mac-1703606497300'
Current agent version: '3.232.0'
Operating System
macOS
11.7.10
20G1427
Runner Image
Image: macos-11
Version: 20231216.1
Included Software: https://github.com/actions/runner-images/blob/macOS-11/20231216.1/images/macos/macos-11-Readme.md
Image Release: https://github.com/actions/runner-images/releases/tag/macOS-11%2F20231216.1
Runner Image Provisioner
2.0.321.1
Current image version: '20231216.1'
```
</details>